File: java-apk/app/src/main/java/come/tshah/app/DialerReceiver.java
package come.tshah.app;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class DialerReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        Intent launch = new Intent(context, MainActivity.class)
            .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
        context.startActivity(launch);
    }
}

File: android/app/src/main/kotlin/come/tshah/app/DialerReceiver.kt
package come.tshah.app

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class DialerReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val launch = Intent(context, MainActivity::class.java).apply {
            addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP)
        }
        context.startActivity(launch)
    }
}
